crates/stc_ts_file_analyzer/src/analyzer/assign/mod.rs

```diff
@@ -1122,7 +1122,22 @@ impl Analyzer<'_, '_> {
             Type::Keyword(KeywordType {
                 kind: TsKeywordTypeKind::TsUnknownKeyword,
                 ..
-            }) => return Ok(()),
+            }) => {
+                // We need to do this because everything is assignable to unknown.
+                // But if we are in argument, we should not allow void (which means no argument
+                // was passed)
+                if self.ctx.in_argument {
```

Comment:
Can we add a field to `AssignOpts` instead of using context?

Reply:
```ts
call(() => {
    var v: void;
    const foo: string = v;
})
```

Reply:
This code is problematic if we use `self.ctx.in_argument`.
We can add a field to `AssignOpts`, e.g. `in_argument: bool` and set it to true while assigning for arguments.

Reply:
I used `allow_assignment_to_void: false` instead, I think that makes sense. I'm not sure why the correct error is reported without me doing anything else tho :thinking: